src/tcgwars/logic/impl/gen7/CelestialStorm.groovy

```diff
@@ -1242,56 +1242,65 @@ public enum CelestialStorm implements LogicCardInfo {
+                    may: true, my.discard, my.all.findAll { !(it.pokemonEX) && !(it.pokemonGX) }
                   )
-                  if (tup[0] == []) break;
+                  if (tup[0] == []) break
                 }
               }
               new Knockout(self).run(bg)
             }
           }
-          move "Electro Ball" , {
+          move "Electro Ball", {
             text "50 damage."
-            energyCost L,C
+            energyCost L, C
             onAttack {
               damage 50
             }
           }
-          move "Crush and Burn GX" , {
+          move "Crush and Burn GX", {
             text "30+ damage. Discard any amount of Energy from your Pokémon. This attack does 50 more damage for each card you discarded in this way. (You can't use more than 1 GX attack in a game.)"
-            energyCost L,C
+            energyCost L, C
             attackRequirement {
               gxCheck()
             }
             onAttack {
               gxPerform()
-              damage 30
-              def bonusDmg = 0
-              while(my.all.findAll{it.cards.energyCount(C)}){
-                if(confirm("discard energy from one pokémon for 50 more damage by energy discarded?")){
-                  def tar = my.all.findAll{it.cards.energyCount(C)}.select("Choose the pokémon to discard energy")
-                  bonusDmg += 50*tar.cards.filterByType(ENERGY).select(min:1,max : tar.cards.filterByType(ENERGY).size()).discard().size()
+              def damageAmount = 30
+              if (my.all.any { it.cards.energyCount() } && confirm("Discard energy from your Pokémon for 50 extra damage per energy discarded?")) {
+                CardList toDiscard = []
+                Map<PokemonCardSet, CardList> workMap = [:]
+                for (PokemonCardSet pcs : my.all) {
+                  if (pcs.cards.filterByType(ENERGY)) workMap.put(pcs, pcs.cards.filterByType(ENERGY))
                 }
-                else{
-                  break
+                PcsList mapTar = workMap.keySet().findAll { workMap.get(it).notEmpty() }
+                while (mapTar) {
+                  PokemonCardSet tar = mapTar.select("Choose the Pokémon to discard energy from. Current Damage: $damageAmount", false)
+                  if (!tar) break
+                  def tarCards = workMap.get(tar).select(min: 0, max: tar.cards.filterByType(ENERGY).size(), "Choose the energies to discard. Current Damage: $damageAmount")
+                  if (!tarCards) break
+                  toDiscard.addAll tarCards
+                  workMap.get(tar).removeAll(tarCards)
+                  damageAmount = 30 + 50 * toDiscard.size()
+                  mapTar = workMap.keySet().findAll { workMap.get(it).notEmpty() }
                 }
+                afterDamage { toDiscard.discard() }
               }
-              damage bonusDmg
+              damage damageAmount
```

Comment:
The important parts of Electrode. The rest is just a bit of formatting.

Reply:
This already took too much time away from school so I'm not going to make any changes currently to it if it seems like it will work. Do make a Trello card for it so when one of us has time we can work on it.